### PR TITLE
Add flag for remove prefix in every kubernetes resources #56

### LINKED ISSUE #56: Update Polymorphic Chart
### Update Polymorphic-App Chart
Add flag for removing prefix (ReleaseName) in every resources

### PR DESCRIPTION
**Pull request by Luv7804**
Closes #56 

## Changes proposed

Add flag for remove prefix in every kubernetes resources

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Review by lalitmoza — APPROVED** [review]

**Review by HussainTechSavvy — APPROVED** [review]

**Review by ankitjilka — APPROVED** [review]

**Review by aniketgohelimp — APPROVED** [review]

File: charts/polymorphic-app/templates/cronjob.yaml [hunk 1]
```diff
@@ -3,10 +3,18 @@
...
+  name: "{{ .name | default $.Values.cronJobTemplate.name }}"
+{{- end }}
   labels:
     {{- include "polymorphic-app.labels" $ | nindent 4 }}
-    app.kubernetes.io/component: {{ $.Release.Name }}-{{ .name | default $.Values.cronJobTemplate.name }}
+{{- if $.Values.prefixWithReleaseName.enabled }}
+    app.kubernetes.io/component: "{{ $.Release.Name }}-{{ .name | default $.Values.cronJobTemplate.name }}"
+{{- else}}
+    app.kubernetes.io/component: "{{ .name | default $.Values.cronJobTemplate.name }}"
+{{- end }}
```
Comment (deltasquare4) [inline charts/polymorphic-app/templates/cronjob.yaml:17]: I think we can make a hard change here. No conditions required.

Comment (deltasquare4) [inline charts/polymorphic-app/templates/cronjob.yaml:17]:
```suggestion
    app.kubernetes.io/component: "{{ .name | default $.Values.cronJobTemplate.name }}"
```

File: charts/polymorphic-app/templates/job.yaml [hunk 2]
```diff
@@ -3,10 +3,18 @@
...
+  name: "{{ .name | default $.Values.jobTemplate.name }}"
+{{- end }}
   labels:
     {{- include "polymorphic-app.labels" $ | nindent 4 }}
-    app.kubernetes.io/component: {{ $.Release.Name }}-{{ .name | default $.Values.jobTemplate.name }}
+        {{- if $.Values.prefixWithReleaseName.enabled }}
+    app.kubernetes.io/component: "{{ $.Release.Name }}-{{ .name | default $.Values.jobTemplate.name }}"
+        {{- else}}
+    app.kubernetes.io/component: "{{ .name | default $.Values.jobTemplate.name }}"
+        {{- end }}
```
Comment (deltasquare4) [inline charts/polymorphic-app/templates/job.yaml:17]: Same as above

File: charts/polymorphic-app/templates/job.yaml [hunk 3]
```diff
@@ -17,7 +25,11 @@ spec:
     metadata:
       labels:
         {{- include "polymorphic-app.labels" $ | nindent 8 }}
-        app.kubernetes.io/component: {{ $.Release.Name }}-{{ .name | default $.Values.jobTemplate.name }}
+        {{- if $.Values.prefixWithReleaseName.enabled }}
+        app.kubernetes.io/component: "{{ $.Release.Name }}-{{ .name | default $.Values.jobTemplate.name }}"
+        {{- else}}
+        app.kubernetes.io/component: "{{ .name | default $.Values.jobTemplate.name }}"
+        {{- end }}
```
Comment (deltasquare4) [inline charts/polymorphic-app/templates/job.yaml:32]: Same as above

File: charts/polymorphic-app/templates/service.yaml [hunk 4]
```diff
@@ -3,10 +3,18 @@
...
+  name: "{{ .name | default $.Values.serviceTemplate.name }}"
+{{- end }}
   labels:
     {{- include "polymorphic-app.labels" $ | nindent 4 }}
-    app.kubernetes.io/component: {{ $.Release.Name }}-{{ .name | default $.Values.serviceTemplate.name }}-svc
+    {{- if $.Values.prefixWithReleaseName.enabled }}
+    app.kubernetes.io/component: "{{ $.Release.Name }}-{{ .name | default $.Values.serviceTemplate.name }}-svc"
+    {{- else }}
+    app.kubernetes.io/component: "{{ .name | default $.Values.serviceTemplate.name }}"
+    {{- end }}
```
Comment (deltasquare4) [inline charts/polymorphic-app/templates/service.yaml:17]: Same as above

File: charts/polymorphic-app/templates/service.yaml [hunk 5]
```diff
@@ -22,17 +30,29 @@ spec:
...
   {{- end }}
   selector:
     matchLabels:
       {{- include "polymorphic-app.selectorLabels" $ | nindent 6 }}
+      {{- if $.Values.prefixWithReleaseName.enabled }}
       app.kubernetes.io/component: "{{ $.Release.Name }}-{{ .name | default $.Values.serviceTemplate.name }}-svc"
+      {{- else}}
+      app.kubernetes.io/component: "{{ .name | default $.Values.serviceTemplate.name }}"
+      {{- end }}
```
Comment (deltasquare4) [inline charts/polymorphic-app/templates/service.yaml:46]: Same as above

File: charts/polymorphic-app/templates/service.yaml [hunk 6]
```diff
@@ -22,17 +30,29 @@ spec:
...
   template:
     metadata:
       labels:
         {{- include "polymorphic-app.labels" $ | nindent 8 }}
-        app.kubernetes.io/component: {{ $.Release.Name }}-{{ .name | default $.Values.serviceTemplate.name }}-svc
+        {{- if $.Values.prefixWithReleaseName.enabled }}
+        app.kubernetes.io/component: "{{ $.Release.Name }}-{{ .name | default $.Values.serviceTemplate.name }}-svc"
+        {{- else}}
+        app.kubernetes.io/component: "{{ .name | default $.Values.serviceTemplate.name }}"
+        {{- end }}
```
Comment (deltasquare4) [inline charts/polymorphic-app/templates/service.yaml:55]: Same as above

File: charts/polymorphic-app/templates/service.yaml [hunk 7]
```diff
@@ -177,7 +201,11 @@ spec:
               labelSelector:
                 matchLabels:
                   {{- include "polymorphic-app.labels" $ | nindent 18 }}
+                  {{- if $.Values.prefixWithReleaseName.enabled }}
                   app.kubernetes.io/component: "{{ $.Release.Name }}-{{ .name | default $.Values.serviceTemplate.name }}-svc"
+                  {{- else}}
+                  app.kubernetes.io/component: "{{ .name | default $.Values.serviceTemplate.name }}"
+                  {{- end }}
```
Comment (deltasquare4) [inline charts/polymorphic-app/templates/service.yaml:208]: Same as above

File: charts/polymorphic-app/templates/service.yaml [hunk 8]
```diff
@@ -201,15 +229,27 @@ spec:
...
+  name: "{{ .name | default $.Values.serviceTemplate.name }}"
+{{- end }}
   labels:
     {{- include "polymorphic-app.labels" $ | nindent 4 }}
-    app.kubernetes.io/component: {{ $.Release.Name }}-{{ .name | default $.Values.serviceTemplate.name }}-svc
+    {{- if $.Values.prefixWithReleaseName.enabled }}
+    app.kubernetes.io/component: "{{ $.Release.Name }}-{{ .name | default $.Values.serviceTemplate.name }}-svc"
+    {{- else}}
+    app.kubernetes.io/component: "{{ .name | default $.Values.serviceTemplate.name }}"
+    {{- end }}
```
Comment (deltasquare4) [inline charts/polymorphic-app/templates/service.yaml:243]: Same as above

File: charts/polymorphic-app/templates/service.yaml [hunk 9]
```diff
@@ -201,15 +229,27 @@ spec:
...
 spec:
   scaleTargetRef:
     apiVersion: apps/v1
     kind: {{ if eq .type "StatefulSet"}}StatefulSet{{- else }}Deployment{{- end }}
+  {{- if $.Values.prefixWithReleaseName.enabled }}
     name: "{{ $.Release.Name }}-{{ .name | default $.Values.serviceTemplate.name }}-svc"
+  {{- else}}
+    name: "{{ .name | default $.Values.serviceTemplate.name }}"
+  {{- end }}
```
Comment (deltasquare4) [inline charts/polymorphic-app/templates/service.yaml:252]: Same as above

File: charts/polymorphic-app/templates/service.yaml [hunk 10]
```diff
@@ -232,10 +272,18 @@ spec:
...
+  name: "{{ .name | default $.Values.serviceTemplate.name }}"
+{{- end }}
   labels:
     {{- include "polymorphic-app.labels" $ | nindent 4 }}
-    app.kubernetes.io/component: {{ $.Release.Name }}-{{ .name | default $.Values.serviceTemplate.name }}-svc
+    {{- if $.Values.prefixWithReleaseName.enabled }}
+    app.kubernetes.io/component: "{{ $.Release.Name }}-{{ .name | default $.Values.serviceTemplate.name }}-svc"
+    {{- else}}
+    app.kubernetes.io/component: "{{ .name | default $.Values.serviceTemplate.name }}"
+    {{- end }}
```
Comment (deltasquare4) [inline charts/polymorphic-app/templates/service.yaml:286]: Same as above

File: charts/polymorphic-app/templates/service.yaml [hunk 11]
```diff
@@ -251,18 +299,25 @@ spec:
 {{- end }}
   selector:
     {{- include "polymorphic-app.selectorLabels" $ | nindent 4 }}
-    app.kubernetes.io/component: {{ $.Release.Name }}-{{ .name | default $.Values.serviceTemplate.name }}-svc
+    {{- if $.Values.prefixWithReleaseName.enabled }}
+    app.kubernetes.io/component: "{{ $.Release.Name }}-{{ .name | default $.Values.serviceTemplate.name }}-svc"
+    {{- else}}
+    app.kubernetes.io/component: "{{ .name | default $.Values.serviceTemplate.name }}"
+    {{- end }}
```
Comment (deltasquare4) [inline charts/polymorphic-app/templates/service.yaml:306]: Same as above

File: charts/polymorphic-app/templates/worker.yaml [hunk 12]
```diff
@@ -3,10 +3,18 @@
...
+  name: "{{ .name | default $.Values.workerTemplate.name }}"
+  {{- end }}
   labels:
     {{- include "polymorphic-app.labels" $ | nindent 4 }}
-    app.kubernetes.io/component: {{ $.Release.Name }}-{{ .name | default $.Values.workerTemplate.name }}
+    {{- if $.Values.prefixWithReleaseName.enabled }}
+    app.kubernetes.io/component: "{{ $.Release.Name }}-{{ .name | default $.Values.workerTemplate.name }}"
+    {{- else }}
+    app.kubernetes.io/component: "{{ .name | default $.Values.workerTemplate.name }}"
+    {{- end }}
```
Comment (deltasquare4) [inline charts/polymorphic-app/templates/worker.yaml:17]: same as above

File: charts/polymorphic-app/templates/worker.yaml [hunk 13]
```diff
@@ -17,12 +25,20 @@ spec:
   selector:
     matchLabels:
       {{- include "polymorphic-app.selectorLabels" $ | nindent 6 }}
+    {{- if $.Values.prefixWithReleaseName.enabled }}
       app.kubernetes.io/component: "{{ $.Release.Name }}-{{ .name | default $.Values.workerTemplate.name }}"
+    {{- else }}
+      app.kubernetes.io/component: "{{ .name | default $.Values.workerTemplate.name }}"
+    {{- end }}
   template:
```
Comment (deltasquare4) [inline charts/polymorphic-app/templates/worker.yaml:33]: Same as above

File: charts/polymorphic-app/templates/worker.yaml [hunk 14]
```diff
@@ -17,12 +25,20 @@ spec:
...
     metadata:
       labels:
         {{- include "polymorphic-app.labels" $ | nindent 8 }}
-        app.kubernetes.io/component: {{ $.Release.Name }}-{{ .name | default $.Values.workerTemplate.name }}
+    {{- if $.Values.prefixWithReleaseName.enabled }}
+        app.kubernetes.io/component: "{{ $.Release.Name }}-{{ .name | default $.Values.workerTemplate.name }}"
+    {{- else }}
+        app.kubernetes.io/component: "{{ .name | default $.Values.workerTemplate.name }}"
+    {{- end }}
```
Comment (deltasquare4) [inline charts/polymorphic-app/templates/worker.yaml:41]: Same as above

File: charts/polymorphic-app/templates/worker.yaml [hunk 15]
```diff
@@ -175,15 +195,27 @@ spec:
...
+  name: "{{ .name | default $.Values.workerTemplate.name }}"
+{{- end }}
   labels:
     {{- include "polymorphic-app.labels" $ | nindent 4 }}
-    app.kubernetes.io/component: {{ $.Release.Name }}-{{ .name | default $.Values.workerTemplate.name }}
+{{- if $.Values.prefixWithReleaseName.enabled }}
+    app.kubernetes.io/component: "{{ $.Release.Name }}-{{ .name | default $.Values.workerTemplate.name }}"
+{{- else }}
+    app.kubernetes.io/component: "{{ .name | default $.Values.workerTemplate.name }}"
+{{- end }}
```
Comment (deltasquare4) [inline charts/polymorphic-app/templates/worker.yaml:209]: Same as above

File: charts/polymorphic-app/templates/worker.yaml [hunk 16]
```diff
@@ -195,7 +227,11 @@ spec:
       target:
         apiVersion: apps/v1
         kind: Deployment
+        {{- if $.Values.prefixWithReleaseName.enabled }}
         name: "{{ $.Release.Name }}-{{ .name | default $.Values.workerTemplate.name }}"
+        {{- else }}
+        name: "{{ .name | default $.Values.workerTemplate.name }}"
+        {{- end }}
```
Comment (deltasquare4) [inline charts/polymorphic-app/templates/worker.yaml:234]: Same as above

File: charts/polymorphic-app/templates/worker.yaml [hunk 17]
```diff
@@ -175,15 +195,27 @@ spec:
 apiVersion: autoscaling/v2beta1
 kind: HorizontalPodAutoscaler
```
Comment (deltasquare4) [inline charts/polymorphic-app/templates/worker.yaml:196]: Should we add a condition not to add HPA for StatefulSets?

**Review by deltasquare4 — CHANGES_REQUESTED** [review]

File: charts/polymorphic-app/templates/worker.yaml (hunk 17 above)
Comment (Luv7804) [reply charts/polymorphic-app/templates/worker.yaml:184]: we can't create StatefulSet for worker, so there no need to add condition in worker.yaml

File: charts/polymorphic-app/templates/service.yaml [hunk 18]
```diff
@@ -198,18 +210,23 @@ spec:
       {{- end }}
 {{- if .autoscaling | default $.Values.serviceTemplate.autoscaling }}
 ---
+{{- if ne .type "StatefulSet"}}
```
Comment (deltasquare4) [inline charts/polymorphic-app/templates/service.yaml:213]: This should be combined with line 211.

**Review by deltasquare4 — CHANGES_REQUESTED** [review]

**Review by deltasquare4 — APPROVED** [review]

**Pull request merged**